dmoj/utils/aux_files.py

```diff
@@ -0,0 +1,61 @@
+import os
+import tempfile
+import traceback
+
+from dmoj.error import InternalError
+from dmoj.executors import executors
+from dmoj.executors.compiled_executor import CompiledExecutor
+
+
+def _mktemp(data):
+    tmp = tempfile.NamedTemporaryFile()
+    tmp.write(data)
+    tmp.flush()
+    return tmp
+
+
+def compile_aux_files(filenames, lang=None, compile_time_limit=None):
+    sources = {}
+
+    try:
+        for filename in filenames:
+            with open(filename, 'rb') as f:
+                sources[os.path.basename(filename)] = f.read()
+    except Exception:
+        traceback.print_exc()
+        raise IOError('Could not read source!')
+
+    def find_runtime(languages):
+        for grader in languages:
+            if grader in executors:
+                return grader
+        return None
+
+    use_cpp = any(map(lambda name: os.path.splitext(name)[1] in ['.cpp', '.cc'], filenames))
+    if lang is None:
+        best_choices = ('CPP17', 'CPP14', 'CPP11', 'CPP0X', 'CPP03') if use_cpp else ('C11', 'C')
+        lang = find_runtime(best_choices)
+
+    clazz = executors.get(lang)
+    if not clazz:
+        raise IOError('could not find an appropriate C++ executor')
+
+    clazz = clazz.Executor

```

Comment:
I think this is better called `Executor`. If you really like to call it a class, use `cls`.